src/domain/collection/curatedEvents/__tests__/CuratedEvents.test.tsx

```diff
@@ -11,106 +11,96 @@ import {
-
-  const chunkedEvents: EventListResponse[] = flow([
-    range,
-    map((id: number) => (id + 1).toString()),
-    chunk(PAGE_SIZE),
-    map((eventIdChunk: string[]) =>
-      fakeEvents(
-        eventIdChunk.length,
-        eventIdChunk.map((eventId) => {
-          const eventName = `Event ${eventId}`;
-          eventNames.push(eventName);
-          curatedEvents.push(`http://localhost:3000/fi/event/${eventId}`);
-          return {
-            endTime: '2020-12-12',
-            startTime: '2020-12-12',
-            id: eventId,
-            name: { fi: `Event ${eventId}` },
-          };
-        })
-      )
-    ),
-  ])(eventsCount);
+afterAll(() => {
+  clear();
+});
 
-  const mocks = chunkedEvents.map((eventList) => {
-    return getMocks(
-      eventList.data as EventFieldsFragment[],
-      eventList.data.map((event) => event.id)
-    )[0];
-  });
+afterEach(() => {
+  jest.restoreAllMocks();
+});
 
-  const collection = fakeCollection({
-    curatedEvents,
-  }) as CollectionFieldsFragment;
+[true, false].forEach((EVENTS_HELSINKI_2) => {
+  test(`should show all events (EVENT_HELSINKI_2 feature is ${
+    EVENTS_HELSINKI_2 ? 'on' : 'off'
+  })`, async () => {
+    setFeatureFlags({ EVENTS_HELSINKI_2 });
+    const events = fakeEvents(
+      eventNames.length,
+      eventNames.map((event, index) => ({
+        id: eventIds[index],
+        name: { fi: event },
+      }))
+    );
+    const courses = fakeEvents(
+      courseNames.length,
+      courseNames.map((event, index) => ({
+        id: courseIds[index],
+        name: { fi: event },
+      }))
+    );
+    const eventsData = events.data as EventFieldsFragment[];
+    const coursesData = courses.data as EventFieldsFragment[];
+    const eventMocks = getMocks(eventsData, eventIds, 'event');
+    const courseMocks = getMocks(coursesData, courseIds, 'course');
 
-  return { mocks, collection, eventNames };
-};
+    render(<CuratedEvents collection={collection} />, {
+      mocks: [...eventMocks, ...courseMocks],
+    });
 
-const waitForRequestToComplete = async () => {
-  await waitFor(() => {
-    expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
-  });
-};
+    await waitFor(() => {
+      expect(screen.queryByTestId('loading-spinner')).not.toBeInTheDocument();
+    });
 
-afterAll(() => {
-  clear();
-});
+    // Test that both lists are
 
-test('should show all events', async () => {
-  const events = fakeEvents(
-    eventNames.length,
-    eventNames.map((event, index) => ({
-      id: eventIds[index],
-      name: { fi: event },
-    }))
-  );
-  const eventsData = events.data as EventFieldsFragment[];
-  const mocks = getMocks(eventsData);
+    const eventsList = within(screen.getByTestId(eventsListTestId));
+    eventNames.forEach((eventName) => {
+      expect(eventsList.getByText(eventName)).toBeInTheDocument();
+    });
 
-  render(<CuratedEventList collection={collection} />, {
-    mocks,
-  });
+    await waitFor(() => {
```

Comment:
Do we need another wait for loading-spinner?

Reply:
Good catch, maybe some copy paste leftover